.changeset/gold-carpets-attend.md

```diff
@@ -0,0 +1,5 @@
+---
+"@bigcommerce/catalyst-core": minor 
+---
+
+This refactor changes the structure of our UI components by replacing composability with a prop-based configuration. This change simplifies the use of our components, eliminating the need to build them individually from a composable approach. Additionally, it provides a single location for all class customizations, improving the experience when fully customizing the component. We believe this approach will make it easier to use components correctly and safeguard against incorrect usage. Ultimately, by adopting a prop-based configuration, we aim to achieve full replaceability and simplify theming for our components.
```

Comment:
Do you think we could add an example of changing usage from the old style to the new in this changeset?

Reply:
Updated!